plugins/usd/module/vtkF3DFaceVaryingPolyData.cxx

```diff
@@ -0,0 +1,147 @@
+#include "vtkF3DFaceVaryingPolyData.h"
+
+#include "vtkFloatArray.h"
+#include "vtkInformation.h"
+#include "vtkInformationIntegerKey.h"
+#include "vtkInformationVector.h"
+#include "vtkPointData.h"
+
+vtkStandardNewMacro(vtkF3DFaceVaryingPolyData);
+
+vtkInformationKeyMacro(vtkF3DFaceVaryingPolyData, INTERPOLATION_TYPE, Integer);
+
+//------------------------------------------------------------------------------
+vtkF3DFaceVaryingPolyData::vtkF3DFaceVaryingPolyData() = default;
+
+//------------------------------------------------------------------------------
+vtkF3DFaceVaryingPolyData::~vtkF3DFaceVaryingPolyData() = default;
+
+//------------------------------------------------------------------------------
+int vtkF3DFaceVaryingPolyData::RequestData(vtkInformation* vtkNotUsed(request),
+  vtkInformationVector** inputVector, vtkInformationVector* outputVector)
+{
+  vtkPolyData* input = vtkPolyData::GetData(inputVector[0]->GetInformationObject(0));
+  vtkPolyData* output = vtkPolyData::GetData(outputVector->GetInformationObject(0));
+
+  // early exit if all interpolations are "vertex"
+  vtkPointData* originalPointData = input->GetPointData();
+
+  vtkIdType nbArrays = originalPointData->GetNumberOfArrays();
+
+  bool earlyExit = true;
+
+  for (vtkIdType i = 0; i < nbArrays; i++)
+  {
+    vtkDataArray* originalArray = originalPointData->GetArray(i);
+
+    vtkInformation* info = originalArray->GetInformation();
+    int interpType = info->Get(vtkF3DFaceVaryingPolyData::INTERPOLATION_TYPE());
```

Comment:
do you plan to reuser this outside of USD ? Outside of F3D (for integrating in VTK) ?

We should name the key accordingly.

Reply:
I'm planning to move it later to the VTK extensions. Regarding its integration to VTK itself, I think it's too soon. I'm still ensure this is the best approach (maybe it's possible to do something at the mapper level to avoid point duplication)

Reply:
fair enough.